NPE in HibernateQueryInterceptor

Hey,

This lib looks great, I wanted to try it in my Spring Boot app (https://github.com/Athou/lapinstance) but I get a NullPointerException:

```
Caused by: java.lang.NullPointerException: null
	at com.yannbriancon.interceptor.HibernateQueryInterceptor.getEntity(HibernateQueryInterceptor.java:86) ~[spring-hibernate-query-utils-1.0.0.jar:1.0.0]
	at org.hibernate.internal.SessionImpl.getEntityUsingInterceptor(SessionImpl.java:550) ~[hibernate-core-5.4.10.Final.jar:5.4.10.Final]
	at org.hibernate.loader.Loader.getRow(Loader.java:1601) ~[hibernate-core-5.4.10.Final.jar:5.4.10.Final]
	at org.hibernate.loader.Loader.getRowFromResultSet(Loader.java:745) ~[hibernate-core-5.4.10.Final.jar:5.4.10.Final]
	at org.hibernate.loader.Loader.processResultSet(Loader.java:1008) ~[hibernate-core-5.4.10.Final.jar:5.4.10.Final]
	at org.hibernate.loader.Loader.doQuery(Loader.java:964) ~[hibernate-core-5.4.10.Final.jar:5.4.10.Final]
	at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:354) ~[hibernate-core-5.4.10.Final.jar:5.4.10.Final]
	at org.hibernate.loader.Loader.doList(Loader.java:2815) ~[hibernate-core-5.4.10.Final.jar:5.4.10.Final]
	at org.hibernate.loader.Loader.doList(Loader.java:2797) ~[hibernate-core-5.4.10.Final.jar:5.4.10.Final]
	at org.hibernate.loader.Loader.listIgnoreQueryCache(Loader.java:2629) ~[hibernate-core-5.4.10.Final.jar:5.4.10.Final]
	at org.hibernate.loader.Loader.list(Loader.java:2624) ~[hibernate-core-5.4.10.Final.jar:5.4.10.Final]
...
```

Any idea ?

Thanks !

--- a/pom.xml
+++ b/pom.xml
@@ -5,7 +5,7 @@
 
     <groupId>com.yannbriancon</groupId>
     <artifactId>spring-hibernate-query-utils</artifactId>
-    <version>1.0.0</version>
+    <version>1.0.1</version>
     <packaging>jar</packaging>
 
     <name>spring-hibernate-query-utils</name>

--- a/src/main/java/com/yannbriancon/interceptor/HibernateQueryInterceptor.java
+++ b/src/main/java/com/yannbriancon/interceptor/HibernateQueryInterceptor.java
@@ -12,6 +12,7 @@ import org.springframework.stereotype.Component;
 import java.io.Serializable;
 import java.util.HashSet;
 import java.util.Set;
+import java.util.function.Supplier;
 
 @Component
 @EnableConfigurationProperties(HibernateQueryInterceptorProperties.class)
@@ -19,14 +20,14 @@ public class HibernateQueryInterceptor extends EmptyInterceptor {
 
     private transient ThreadLocal<Long> threadQueryCount = new ThreadLocal<>();
 
-    private transient ThreadLocal<Set<String>> threadPreviouslyLoadedEntities = new ThreadLocal<>();
+    private transient ThreadLocal<Set<String>> threadPreviouslyLoadedEntities =
+            ThreadLocal.withInitial(new EmptySetSupplier());
 
     private static final Logger LOGGER = LoggerFactory.getLogger(HibernateQueryInterceptor.class);
 
     private final HibernateQueryInterceptorProperties hibernateQueryInterceptorProperties;
 
     public HibernateQueryInterceptor(HibernateQueryInterceptorProperties hibernateQueryInterceptorProperties) {
-        threadPreviouslyLoadedEntities.set(new HashSet<>());
         this.hibernateQueryInterceptorProperties = hibernateQueryInterceptorProperties;
     }
 
@@ -115,5 +116,11 @@ public class HibernateQueryInterceptor extends EmptyInterceptor {
             default:
                 throw new NPlusOneQueryException(errorMessage);
         }
+    }
+}
+
+class EmptySetSupplier implements Supplier<Set<String>> {
+    public Set<String> get(){
+        return new HashSet<>();
     }
 }

